pinot-spi/src/main/java/org/apache/pinot/spi/config/table/SegmentZKMetadataConfig.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+// TODO Possibly redundant class. Use SegmentZKMetadata throughout?
```

Comment:
Any Blocker for doing this?

Reply:
Will have to move SegmentZKMetadata out of pinot-commons to pinot-spi. I'll be checking if that could break something. If all is fine, will update the PR

Reply:
Adding to this, segmentZkMeta is not guaranteed to have endOffset during segment creation (only added during segment commit step). Looks like this class would be needed in that case.

Reply:
that would touch a lot of files, and I don't even think you'll be able to do that without breaking some things along the way. So if at all, attempt that in a separate PR

Reply:
Ack. Indeed its a rather non trivial refactor

Reply:
cool! Yeah, I guess there are a couple of interface that need to go in `pinot-spi` instead of core or commons. We can take that up another time.